libdnf5-cli/output/advisorylist.cpp

```diff
@@ -104,18 +107,68 @@ void print_advisorylist_table(
     scols_unref_table(table);
 }
 
+static json_object * adv_pkg_as_json(auto & adv_pkg) {
+    auto advisory = adv_pkg->get_advisory();
+    json_object * json_advisory = json_object_new_object();
+    json_object_object_add(json_advisory, "name", json_object_new_string(advisory->get_name().c_str()));
+    json_object_object_add(json_advisory, "type", json_object_new_string(advisory->get_type().c_str()));
+    json_object_object_add(json_advisory, "severity", json_object_new_string(advisory->get_severity().c_str()));
+    json_object_object_add(json_advisory, "nevra", json_object_new_string(adv_pkg->get_nevra().c_str()));
+    unsigned long long buildtime = advisory->get_buildtime();
+    json_object_object_add(
+        json_advisory, "buildtime", json_object_new_string(libdnf5::utils::string::format_epoch(buildtime).c_str()));
+    return json_advisory;
+}
+
+static json_object * adv_refs_as_json(
+    auto & pretty_reference_type, auto & reference_type, auto & adv_pkg, auto & advisory) {
+    json_object * json_advisory = json_object_new_object();
+    json_object_object_add(json_advisory, "advisory_id", json_object_new_string(advisory.get_name().c_str()));
+    json_object_object_add(json_advisory, "advisory_type", json_object_new_string(advisory.get_type().c_str()));
+    json_object_object_add(json_advisory, "advisory_severity", json_object_new_string(advisory.get_type().c_str()));
+    json_object_object_add(json_advisory, "nevra", json_object_new_string(adv_pkg.get_nevra().c_str()));
+
+    json_object * json_adv_references = json_object_new_array();
+    auto references = advisory.get_references({reference_type});
+    for (auto reference : references) {
+        json_object * json_reference = json_object_new_object();
+        json_object_object_add(json_reference, "id", json_object_new_string(reference.get_id().c_str()));
+        json_object_object_add(json_reference, "type", json_object_new_string(reference_type.c_str()));
+        json_object_object_add(json_reference, "pretty_type", json_object_new_string(pretty_reference_type.c_str()));
+        unsigned long long buildtime = advisory.get_buildtime();
+        json_object_object_add(
+            json_reference,
+            "buildtime",
+            json_object_new_string(libdnf5::utils::string::format_epoch(buildtime).c_str()));
```

Comment:
I just noticed this should be on the advisory rather than on the reference. Fixing.